Initial Live TV Support

neilsb opened the pull request:
Added initial live TV Support (using Code provided by @Musi13)

**Changes**
- Added LiveTV Page (Currently showing Channels)
- Implemented viewing of Live TV Streams 
- Additional Checking for DirectPlay property of Media Source
- Colorizing of post backgrounds for items which have no Post Images
- Increased Server Timeout of POST requests from 5s to 30s
- Inform server of Roku Capabilities after Login

**Issues**
Refs #26

Musi13 commented on source/VideoPlayer.brs:
```diff
@@ -177,6 +207,9 @@ end function
 
 function directPlaySupported(meta as object) as boolean
   devinfo = CreateObject("roDeviceInfo")
+  if meta.json.MediaSources[0] <> invalid or meta.json.MediaSources[0].SupportsDirectPlay = false then
```
Should this be `meta.json.MediaSources[0] = invalid`?

neilsb replied:
I think I probably meant this to be _and_ and instead of an _or_.  Not change any of the existing logic, unless the SupportsDirectPlay exists and is false.

jcdick1 commented:
If commenting on PRs is allowed, I'd like to suggest adding 265 to the video codecs and MKV to the containers, for the Ultra and Premiere models that support 4K in the DirectPlay capabilities patch.

neilsb commented:
> If commenting on PRs is allowed

Of course.  Any feedback is welcome.

> I'd like to suggest adding 265 to the video codecs and MKV to the containers, for the Ultra and Premiere models and TVs that support 4K in the DirectPlay capabilities patch.

We should be able to query if the 265 codec is supported by the device, and reading the [Roku Spec](https://developer.roku.com/en-gb/docs/specs/streaming.md) I think it's implying MKV is supported across the board, so I'll get that added to the list of supported containers.

neilsb commented:
Added mkv and h265 to Direct Play support (as mentioned by @jcdick1) and also added additional some additional checking if Audio Bitrate, and Video Container and Profile when determining if we can Direct Play a source

jcdick1 commented:
So does the new update make DirectPlay of HEVC resolution-specific?  I know I used that as an example of use, but I don't know that it should be resolution specific.  Some folks are using HEVC even at 720p for optimal disk space use.

neilsb commented:
> So does the new update make DirectPlay of HEVC resolution-specific? 

Well spotted.  Yes, the way that I've done it depends on the current video output.

I think from [the specification](https://developer.roku.com/en-gb/docs/specs/streaming.md#supported-video-codecs), HEVC (and VP9 actually) is  only supported on Roku 4K capable devices.  But it could well be a 4k Roku device plugged into a 1080p TV, and that _would_ still support HEVC, so it should be enabled in that instance.

I guess what I should be doing it querying the Roku device to ask it if it supports HEVC and some resolution (i.e. 720p - since we don't necessarily have a video to play when we try to build the profile data) and use the result from that.

jcdick1 commented:
I'm not sure how it all works, but would AC3/E-AC3 need to go in under audio codecs with AAC, Opus, and such?

neilsb commented:
I've had a look at this and talked to the core team a little about this

> I'm not sure how it all works [...]

This seems the be the consensus at the moment.  This area is a bit of a _black box_ right now and has not been dug into since the initial fork.  There are a couple of people looking into it, and I've been able to glean a little more information from the server code, but what is clear is that we need to put a bit of effort into building a proper Device Profile for Roku devices.   (I suspect #225 is also due to the current Device Profile).

I would suggest that we do not try to resolve that in this PR, and instead raise a separate ticket / PR for sorting a proper Profile.

anthonylavado commented:
This works amazingly well with both M3U tuner sources, and my TV Tuner (an antenna, connected to a Hauppauge WinTV-HVR950, connected to TVHeadend, connected to Antennas to emulate an HDHomeRun). Thank you!

jrbski commented:
I just loaded the 1.3.0 release on my Roku TV using the beta link on my Roku account, and I'm now streaming LiveTV from my Jellyfin server using a HDHomerun prime tuner. It's working really well, and the Roku TV is remote (at my cabin) from my  Jellyfin server at home. I had installed the old beta, and I didn't even need to configure my server connection. :)

My first impression is more great work by the Jellyfin team... Thanks for the great work!!
